toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -442,7 +442,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 	defer func() {
 		// graph manipulation calls may panic on error (such as duplicate node IDs)
 		if r := recover(); r != nil {
-			err = fmt.Errorf("collapsing nodes (%v) into (%s) failed, error: %s", nodesToCollapse, versionedPkg, r)
+			err = fmt.Errorf("failed to collapse nodes (%v) into (%s) failed:\n%w\n%s", nodesToCollapse, versionedPkg, err, r)
```

Comment:
If there was a panic, will `err` contain a value?

Reply:
Thanks for pointing this out. defer() would either catch `err` or `panic`, so will modify this to return only the required values.